File: src/main/Pieces/Bishops/Bishop.cpp
//
// Created by Andrea on 26/01/2022.
//

#include "Bishop.h"
#include <iostream>


Bishop::Bishop(bool own, string name, Position p) : Piece(own, move(name), p) { };


vector<Position>* Bishop::validMove(Position start, Position end) {

    vector<Position>* trace = Piece::validMove(start, end);

    if (trace){

        int colStart = start.getRealCoord().first;
        int rowStart = start.getRealCoord().second;
        int colEnd = end.getRealCoord().first;
        int rowEnd = end.getRealCoord().second;

        int diffCol = abs(colStart - colEnd);
        int diffRow = abs(rowStart - rowEnd);

        if (diffCol == diffRow) {

            Piece::diagonalMovement(start, end, trace);
            return trace;
        }

    }
    return nullptr;

}

File: ChessApp/src/Pieces/Bishop/Bishop.cs
using ChessApp.src.Positions;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ChessApp.src.Pieces.Bishop
{
    internal class Bishop: Piece
    {


        public Bishop(bool own, string name, Position p) : base(own, name, p) { }

        public override List<Position>? validMove(Position start, Position end)
        {
            List<Position>? trace = base.validMove(start, end);

            if (trace != null)
            {

                int colStart = start.getRealCoord().first;
                int rowStart = start.getRealCoord().second;
                int colEnd = end.getRealCoord().first;
                int rowEnd = end.getRealCoord().second;

                int diffCol = abs(colStart - colEnd);
                int diffRow = abs(rowStart - rowEnd);

                if (diffCol == diffRow)
                {

                    Piece.diagonalMovement(start, end, trace);
                    return trace;
                }

            }
            return null;
        }

        

    }
}
